src/components/burger-constructor/burger-constructor.jsx

```diff
@@ -1,15 +1,31 @@
 import React from "react";
-import { ConstructorElement, CurrencyIcon, Button, DragIcon } from "@ya.praktikum/react-developer-burger-ui-components";
-import styles from "./burger-constructor.module.css";
 import PropTypes from "prop-types";
-import Modal from "../modal/modal";
+
+import { ConstructorElement, CurrencyIcon, Button, DragIcon } from "@ya.praktikum/react-developer-burger-ui-components";
 import OrderDetails from "../order-details/order-details";
+import Modal from "../modal/modal";
+import IngredientPropTypes from "../../utils/utils";
+
+import styles from "./burger-constructor.module.css";
 
 const BurgerConstructor = (props) => {
   const [modalActive, setModalActive] = React.useState(false);
-
   const bun = props.data.find((item) => item.type === "bun");
-  let totalPrice = bun?.price;
+  
+  const mainsAndSouses = [];
+
+  props.data.map((ingredient) => {
+    if (ingredient.type !== "bun") {
+      mainsAndSouses.push(ingredient);
+      return mainsAndSouses;
+    }
+    return true;
+  });
```

Comment:
Можно лучше:
тут отлично пойдет метод `filter`, и можно будет не мутировать созданный массив
```suggestion
  const mainsAndSouses = data.filter(({ type }) => type !== 'bun');
```